etsin_finder/qvain_light_rpc.py

```diff
@@ -0,0 +1,86 @@
+# This file is part of the Etsin service
+#
+# Copyright 2017-2018 Ministry of Education and Culture, Finland
+#
+# :author: CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
+# :license: MIT
+
+"""RPC API endpoints, meant to be used by Qvain Light form"""
+
+from functools import wraps
+import inspect
+from flask import request, session
+from flask_restful import abort, reqparse, Resource
+
+from etsin_finder.app_config import get_app_config
+from etsin_finder import authentication
+from etsin_finder.qvain_light_service import change_cumulative_state
+from etsin_finder.finder import app
+from etsin_finder.qvain_light_utils import get_dataset_creator
+
+log = app.logger
+
+def log_request(f):
+    """
+    Log request when used as decorator.
+
+    :param f:
+    :return:
+    """
+    @wraps(f)
+    def func(*args, **kwargs):
+        """
+        Log requests.
+
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        csc_name = authentication.get_user_csc_name() if not app.testing else ''
+        log.info('[{0}.{1}] {2} {3} {4} USER AGENT: {5}'.format(
```

Comment:
This might add duplicate fields to the logs but this could be resolved after merge if needed.